Add a post about installing feature packs with Prospero

spyrkob opened the pull request.

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+
+== Additional feature packs for Wildfly
+
+Wildfly server capabilities can be extended by installing additional feature packs such as wildfly-datasources-galleon-pack or wildfly-myfaces-galleon-pack.
```
could you link to https://github.com/wildfly-extras/wildfly-datasources-galleon-pack & https://github.com/wildfly-extras/wildfly-myfaces-feature-pack?

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+== Additional feature packs for Wildfly
+
+Wildfly server capabilities can be extended by installing additional feature packs such as wildfly-datasources-galleon-pack or wildfly-myfaces-galleon-pack.
+The feature packs can be installed using Galleon or Prospero.
```
idem with https://github.com/wildfly-extras/prospero & https://docs.wildfly.org/galleon/

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+
+== Prerequisites
+
+1. Download and unzip Wildfly 29.0.1.Final from https://www.wildfly.org/downloads/
```
We can also directly link to https://github.com/wildfly/wildfly/releases/download/29.0.1.Final/wildfly-29.0.1.Final.zip

jmesnil replied:
I assume from the rest of the article that the server must be copied to `/tmp/wildfly`. If that's the case, that better to specify it now. Or use the instructions to do so with the `cd /tmp/` & `wget`.

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+== Prerequisites
+
+1. Download and unzip Wildfly 29.0.1.Final from https://www.wildfly.org/downloads/
+1. Checkout Prospero from https://github.com/wildfly-extras/prospero
```
wildfly.org now supports copy/paste to clipboard. For instructions that users have to run, that's now better to provide the actual instructions with something like:

```
[source]
----
git clone https://github.com/wildfly-extras/prospero
cd prospero
mvn install -Pdist
----
```

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+
+ ----
+# wildfly
+  manifest: file:/tmp/wildfly-29.0.1.Final/.installation/manifest-wildfly-29.0.1.Final.yaml
```
is the server in `/tmp/wildfly-29.0.1.Final/`or in `/tmp/wildfly/`

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+" > /tmp/datasources-manifest.yaml
+----
+
+Then we can register Wildfly to receive updates from a new channel:
```
s/Wildfly/WildFly

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+
+== Updating mariadb driver version
+
+Steps above installed a default version of MariaDB driver, but using prospero and wildfly channels we are able to udate that.
```
s/wildfly/WildFly

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+
+Steps above installed a default version of MariaDB driver, but using prospero and wildfly channels we are able to udate that.
+
+First, lets add desired version of the driver to the datasources manifest:
```
let's

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+author: spyrkob
+description: Using Prospero to installing additional feature packs in existing Wildfly servers.
+---
+
```
@spyrkob you're adding great content describing Prospero. We should provide a breadcrumb across all the articles.

For example, you could add a short introduction that starts with: 

```
In [https://www.wildfly.org/news/2023/04/05/prospero/](https://www.wildfly.org/news/2023/04/05/prospero/), you learnt how to install and use [https://github.com/wildfly-extras/prospero](Prospero) to provision a WildFly server. In this article, you will learn how to use Prospero to modify an existing server and extends its capabilities by adding a MariaDB datasource.
```

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+Wildfly server capabilities can be extended by installing additional feature packs such as wildfly-datasources-galleon-pack or wildfly-myfaces-galleon-pack.
+The feature packs can be installed using Galleon or Prospero.
+
+In this article we will use Prospero to install a MariaDB datasource provided by wildfly-datasources-galleon-pack.
```
related to my "breadcrumb" comment above, insists that it's installing to an existing unzipped download of WildFly. something like: "install a MariaDB datasource provided by wildfly-datasources-galleon-pack to an existing WildFly server (downloaded from the server zip archive".

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+
+[,bash]
+----
+./prospero.sh channel list --dir /tmp/wildfly
```
here again, if you could split the command to run `prospero.sh channel list --dir /tmp/wildfly` from its output , the user could directly paste it. THat's a pattern we should do in our code snippet.

```
You can run the `update subscribe` command to generate a manifest based on selected version of WildFly server and register the server to a channel using this manifest:
[source]
----
./prospero.sh channel list --dir /tmp/wildfly
----

The command will output the channels that are the server is subscribed to:

[source]
----
Server /tmp/wildfly is subscribed to following channels:

---
# wildfly
  manifest: file:/tmp/wildfly-29.0.1.Final/.installation/manifest-wildfly-29.0.1.Final.yaml
  repositories:
    id: central
    url: https://repo1.maven.org/maven2/
    id: jboss-public
    url: https://repository.jboss.org/nexus/content/groups/public/
    id: mrrc
    url: https://maven.repository.redhat.com/ga/
---
----
```

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+
+[,bash]
+----
+./prospero channel add \
```
idem about splitting the command execution and its output

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,149 @@
+----
+ls /tmp/wildfly/modules/org/mariadb/jdbc/main
+mariadb-java-client-3.2.0.jar  module.xml
+----
```
Is it interesting to see a snippet of the output of `/tmp/wildfly/bin/standalone.sh`?

Could you please add a Conclusion that summarise what we learnt from this article.

jmesnil reviewed (changes requested):
@spyrkob thanks! I made a few suggestions to improve the flow of the article but the technical content is great!

jmesnil commented:
@spyrkob that's beyond the scope of this article but it would be great to be able to use prospero without having to build it from source. Could we add an executable binary whenever a new release of Prospero is created at https://github.com/wildfly-extras/prospero/releases?

spyrkob commented:
@jmesnil thanks for the review, I updated the article and added a prospero release that the binary can be downloaded from

spyrkob commented:
@jmesnil when you have a moment could you re-review this blog post please?

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+---
+layout: post
+title:  "Installing additional feature packs in WildFly"
+date:   2023-09-13
```
will have to be updated

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+
+== Introduction
+
+In a previous article https://www.wildfly.org/news/2023/04/05/prospero/, you learnt how to install 
```
https://www.wildfly.org/news/2023/04/05/prospero/[previous article]

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+
+== Additional feature packs for WildFly
+
+Galleon feature packs provide a mechanism to extend WildFly server capabilities. An example of feature packs offered by the WildFly community are 
```
s/offered/contributed ?

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+
+== Prerequisites
+
+1. Download WildFly 29.0.1.Final and unzip it in `/tmp` folder:
```
should we update to WildFly 30?

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+streams:
+  - groupId: org.wildfly
+    artifactId: wildfly-datasources-galleon-pack
+    version: 5.0.0.Final
```
6.0.0.Final has been released last week

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+
+== Updating mariadb driver version
+
+Steps above installed a default version of MariaDB driver, but using prospero and wildfly channels we are able to udate that.
```
s/wildfly/WildFly

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+
+The detailed description of available configuration options can be found in the datasource feature pack documentation, 
+eg. https://github.com/wildfly-extras/wildfly-datasources-galleon-pack/blob/main/doc/mariadb/README.md#required-configuration[MariaDB configuration].
+Managing WildFly system properties is described in the https://docs.wildfly.org/16/Admin_Guide.html#properties-2[WildFly Admin Guide]
```
let's link to https://docs.wildfly.org/30/Admin_Guide.html instead :)

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+
+[source]
+----
+export MARIADB_USER=test
```
it might be worth noting that we did nothing about the host as `localhost:3306` is the default

jmesnil commented on _posts/2023-09-13-prospero-add-feature-pack.adoc:
```diff
@@ -0,0 +1,269 @@
+In this article, we learned how to use Prospero to extend WildFly server with additional capabilities using feature packs.
+
+In particular we used https://github.com/wildfly-extras/wildfly-datasources-galleon-pack[wildfly-datasources-galleon-pack] to install 
+and update a MariaDB datasource, but the same technique can be used to install other feature packs provided by the community.
```
s/provided/contributed

jmesnil reviewed (changes requested):
@spyrkob sorry, some of the versions are outdated as I've been slow to review your PR. 

Otherwise, I proposed a few changes but the overall article looks good.

spyrkob commented:
@jmesnil sorry, I should have updated the versions earlier. Renamed the blog file, and updated the versions to work with wildfly 30.0.0.Final/wildfly-datasources-galleon-pack 6.0.0.Final

jmesnil commented:
@spyrkob thanks!
